app/models/ems_event.rb

```diff
@@ -136,6 +137,18 @@ def self.process_host_in_event!(event, options = {})
     end
   end
 
+  def self.process_physical_storage_in_event!(event, options = {})
+    storage_system_uid = event[:full_data][:storage_system]
+    process_object_in_event!(PhysicalStorage, event, options)
+
+    if event[:physical_storage_id].nil? && storage_system_uid.present?
+      storage = PhysicalStorage.find_by(:ems_ref => storage_system_uid)
+      unless storage.nil?
+        event[:physical_storage_id] = storage.id
+      end
+    end
```

Comment:
Hey @TonySchneider I believe if you set `event[:physical_storage_ems_ref]` in your event parser (https://github.com/ManageIQ/manageiq-providers-autosde/blob/master/app/models/manageiq/providers/autosde/storage_manager/event_catcher/runner.rb#L34-L44) then `process_object_in_event` will do this all properly.

We don't want to have to check the `event[:full_data]` here because in core this has to work for any physical provider where `event[:full_data]` is provider specific.

Reply:
You wrote that it removed all keys that end with '_ems_ref' or am I wrong?
@agrare 
I can call it :physical_storage_uid or something..

Reply:
Oh sorry for the confusion, it removes all keys that end with `_ems_ref` _after_ this processing and before saving.  The intention is so that you can use them in the payload without having to have columns for them in the table which is exactly what we're wanting here.

Reply:
right, it happens after this method
https://github.com/ManageIQ/manageiq/blob/41e0d0beaf44b8eba8fd47eb9d3749895156b975/app/models/ems_event.rb#L226

on it